### PR TITLE
Split threat model for log4j and log4net

### PR DESCRIPTION
**Pull request by FreeAndNil**
splitted threat model for log4j and log4net

### CODE REVIEW

File: _threat-model-common.adoc
Comment (vy) [inline _threat-model-common.adoc:1]: @FreeAndNil, please keep `-common.adoc`. Logging Services project delivers several software (mostly written in Java), not just Log4j and Log4Net. I see that certain issues here do not apply to Log4Net. Instead of nuking the file, let's remark those exceptions explicitly. I will point these out in other threads.

File: _threat-model-common.adoc [hunk 1]
```diff
@@ -1,44 +0,0 @@
...
-
-[#threat-common-config-sources]
-==== Configuration sources
-All configuration sources to an application must be trusted by the programmer.
-When loading a configuration file from disk (especially when a monitor interval is configured to reload the file periodically), the location of the configuration file must be kept safe from unauthorized modifications.
-Similarly, when loading a configuration file over the network such as through HTTP, this should be configured to use TLS or a secure connection in general with strong authentication guarantees.
-This remote location must be kept safe from unauthorized modifications.
-When configurations are modified through JMX, the JMX server should be safely configured to require authentication and a secure connection if being accessed over the network.
-When configurations are provided through JNDI, these should only use the `java` scheme for sharing configurations in a Java EE or Jakarta EE application service.
-JNDI-sourced configurations should not use other JNDI providers such as LDAP, DNS, or RMI, as all these providers are difficult to properly secure.
```
Comment (vy) [inline _threat-model-common.adoc]: You can place these last two sentences to a new paragraph starting with:

```

For Java-based projects supporting JNDI, when configurations are provided ...
JNDI-sourced configurations ...
```

File: _threat-model-common.adoc [hunk 2]
```diff
@@ -1,44 +0,0 @@
...
-
-[#threat-common-java-serialization]
-==== Java Object Serialization Stream Protocol
```
Comment (vy) [inline _threat-model-common.adoc]: We can keep this in `-common.adoc` – it is clear that it has nothing to do with .NET environment.

File: _threat-model-log4j.adoc
Comment (vy) [inline _threat-model-log4j.adoc:1]: When `-common.adoc` is back, these changes can be reverted.

File: _threat-model-log4net.adoc [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+Information on how to verify releases signatures are explained further in xref:download.adoc[the Download page].
+Thus, GPG signatures should be validated in your build process.
+
+[#threat-log4net-config-sources]
+==== Configuration sources
+All configuration sources to an application must be trusted by the programmer.
+When loading a configuration file from disk (especially when a monitor interval is configured to reload the file periodically), the location of the configuration file must be kept safe from unauthorized modifications.
+Similarly, when loading a configuration file over the network such as through HTTP, this should be configured to use TLS or a secure connection in general with strong authentication guarantees.
+This remote location must be kept safe from unauthorized modifications.
+
```
Comment (vy) [inline _threat-model-log4net.adoc]: When `-common.adoc` is back, these sections can be removed.

File: _threat-model-log4net.adoc [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+[#threat-log4net-parametrized-logging]
+==== Parameterized logging
+
+When using a log message containing template parameters like `{0}`, only the format string is evaluated for parameters to be substituted.
+The message parameters themselves are not evaluated for parameters; they are only included in the format string corresponding to their template position.
+The conversion of message parameters into a string is done on-demand depending on the layout being used.
+When structure-preserving transformations of log message data are required, the `Message` API should be used for logging structured data combined with a structured layout (e.g., `XmlLayout`).
```
Comment (vy) [inline _threat-model-log4net.adoc]: Does Log4Net have a `Message` API? (`Log masking` section below also has a mention of it.)

File: _threat-model-log4net.adoc [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+As such, when using unstructured layouts, no user-controlled input should be included in logs.
+It is strongly recommended that a structured layout (e.g., `XmlLayout`) is used instead for these situations.
+
+[#threat-log4net-structured-logging]
+==== Structured logging
+
+When using a structured layout (most layouts besides pattern layout), log messages are encoded according to various output formats.
+These safely encode the various fields included in a log message.
+For example, the `XmlLayout` can be used to output log messages in an XML structure where all log data is properly encoded into safely parseable XML.
+This is the recommended mode of operation for use with log parsing and log collection tools that rely on log files or arbitrary output streams.
```
Comment (vy) [inline _threat-model-log4net.adoc:47]: Is this statement practically true? That is, in .NET world, do log collection tools work with XML-based log files?

**Review by vy — CHANGES_REQUESTED** [review]

**Comment by ppkarwasz** [comment]
@FreeAndNil,

I think it would be better to move as much content as possible to the "Common" security threat model.

File: _threat-model-log4net.adoc (hunk 4 above)
Comment (FreeAndNil) [reply _threat-model-log4net.adoc]: Yes, as mentioned in the PMC channel, you can call
`logger.Warn(object? message)`
Log4Net calls .ToString() for the "message" to serialize it.

File: _threat-model-common.adoc
Comment (FreeAndNil) [reply _threat-model-common.adoc:1]: check

File: _threat-model-common.adoc (hunk 2 above)
Comment (FreeAndNil) [reply _threat-model-common.adoc]: check

File: _threat-model-log4net.adoc (hunk 3 above)
Comment (FreeAndNil) [reply _threat-model-log4net.adoc]: check

File: _threat-model-log4net.adoc (hunk 5 above)
Comment (FreeAndNil) [reply _threat-model-log4net.adoc:47]: Yes. See https://discuss.elastic.co/t/parsing-xml-log-file-with-logstash/167306 and https://stackoverflow.com/questions/51382827/parse-complex-xml-file-using-elk

**Comment by FreeAndNil** [comment]
@ppkarwasz, @vy I've applied the changes you requested.

File: _threat-model-log4net.adoc (hunk 4 above)
Comment (vy) [reply _threat-model-log4net.adoc]: I don't see a dedicated type/contract, but a generic `object` (contrary to Log4j, which uses the `Message` interface, hence the wording), am I mistaken?

Log4j's `Message` type provides contract(s) on how shall its content be encoded for a certain formatting; JSON, XML, etc. AFAIU, Log4Net does not have such a type and calling `.ToString()` is not structure-preserving. I'd reword this sentence as follows:

```suggestion
When structure-preserving transformations of log data are required, a structured layout (e.g., `XmlLayout`) should be used.
```

File: _threat-model-log4net.adoc [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+[#threat-log4net-log-masking]
+==== Log masking
+
+Log4Net, like any other generic logging library, cannot generically support log masking of sensitive data.
+While custom plugins may be developed to attempt to mask various regular expressions (such as a string that looks like a credit card number), the general problem of log masking is equivalent to the halting problem in computer science where sensitive data can always be obfuscated in such a way as to avoid detection by log masking.
+As such, it is the responsibility of the developer to properly demarcate sensitive data such that it can be consistently masked by log masking plugins.
+This sort of use case should make use of the `Message` API for better control over the output of such data.
```
Comment (vy) [inline _threat-model-log4net.adoc]: Due to absence of a `Message` API, I'd remove this suggestion.

```suggestion
```

Comment (FreeAndNil) [reply _threat-model-log4net.adoc]: Thanks.

File: _threat-model-log4net.adoc (hunk 5 above)
Comment (vy) [reply _threat-model-log4net.adoc:47]: Of course ELK can ingest anything, including XML. I am just curious how common is this. I have never seen and/or heard any IT system delivering logs in XML. Is this a common practice in the .NET world?

**Review by vy — APPROVED** [review]
LGTM. Thanks @FreeAndNil!

**Review by ppkarwasz — APPROVED** [review]
LGTM
